### PR TITLE
Adding test cases for eval error_rate and revert to normal tuple handling …

### LINKED ISSUE #1598: Test cases needed after eval error_rate logic merges
Need to test:
- Error rate of 0 for mt_bench and mt_bench_branch
- Error rate of > 0 for mt_bench and mt_branch_branch

Code changes:
- Fix TODOs for temporary number of return values handling from evaluate.py

Depends on:

https://github.com/instructlab/instructlab/pull/1572/files
https://github.com/instructlab/eval/pull/49
https://github.com/instructlab/instructlab/pull/1597

### PR DESCRIPTION
**Pull request by danmcp**
This PR adds unit test cases for the new error rate logic from the eval library.  There is one set of logic for a 0 error rate (which doesn't display anything new) and a separate check for when an error_rate > 0 exists.

**Issue resolved by this Pull Request:**
Resolves #1598 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by danmcp** [comment]
Note: The unit tests will fail until intructlab-eval 0.0.6 is released

**Comment by danmcp** [comment]
Please don't merge until instruct-eval 0.0.6 is released

File: requirements.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ GitPython>=3.1.42
 # see https://github.com/instructlab/instructlab/issues/579
 # bitsandbytes; sys_platform=='linux' and platform_machine=='x86_64'
 httpx>=0.25.0
-instructlab-eval>=0.0.4
+instructlab-eval>=0.0.8
```
Comment (cdoern) [inline requirements.txt:12]: large bump, is this the newest version?

**Review by cdoern — COMMENTED** [review]
general comments before approving

File: requirements.txt (hunk 1 above)
Comment (alinaryan) [reply requirements.txt:12]: yes

**Review by alinaryan — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**